#161255373 Article Reports

AnguleMathias opened the pull request:
#### What does this PR do?
* Authenticated users should be able to report an article

#### Description of Task to be completed?
* As an authenticated user I should be able to report articles 

#### How should this be manually tested?
* Reporting an article: `/api/articles/<str:article_slug>/report/`

#### Related Pivotal Tracker story
#161255373 -[Users should be able to report an article](https://www.pivotaltracker.com/story/show/161255373)

#### Screenshots
<img width="1356" alt="screenshot 2018-11-14 at 11 51 27" src="https://user-images.githubusercontent.com/39617195/48470853-a70d1b00-e803-11e8-8266-84c49b697e84.png">

AnguleMathias committed:
Feature(Article Reports): Users should be able to report an article

- Added report to models
- Added report method to comment serializer class
- Added report endpoint
- Tests for report funtionality

[Feature #161255373]

AnguleMathias committed:
Feature(Article Reports): Users should be able to report an article

- Added report to models
- Added report method to comment serializer class
- Added report endpoint
- Tests for report funtionality

[Feature #161255373]

AnguleMathias committed:
Feature(Article Reports): Users should be able to report an article

- Update .gitignore

[Feature #16125537]

AnguleMathias committed:
Feature(Article Reports): Users should be able to report an article

- Update .gitignore

[Feature #16125537]

Jwesh reviewed (approved):
lgtm

maxwellgithinji reviewed (approved):
LGTM

njeri-ngigi commented on authors/apps/articles/tests/test_article_report.py:
```diff
@@ -0,0 +1,73 @@
+from rest_framework.views import status
+
+
+class ReportArticleTestCase(BaseTest):
```
In base.py there is an author_user and non_author_user defined. Since you've inherited them you can use self.token or self.non_user_token for the token rather than defining another method for that.

AnguleMathias replied:
from base .py I am making use the URL's for login and sign up as well as article_data the user_data so as to get the token as a string.

njeri-ngigi commented on authors/apps/articles/tests/test_article_report.py:
```diff
@@ -0,0 +1,73 @@
+        return token
+
+    def create_article(self, token, article):
+        """ Method to create an article"""
```
also in base.py there's an article posted already with the slug test-title you can make use of that too.

njeri-ngigi commented on authors/apps/articles/urls.py:
```diff
@@ -37,4 +37,5 @@
     path('articles/<slug>/comment/<int:id>/',
          comments_list, name='comment_an_article'),
     path('articles/<slug>/comment/', comments_details, name='modify_a_comment'),
+    path('articles/<slug>/report', ReportCreateAPIView.as_view()),
```
Do we have a standard for urls? All other urls have a trailing slash

AnguleMathias replied:
corrected

3N61N33R commented on authors/apps/articles/models.py:
```diff
@@ -56,3 +56,15 @@ class Comment(models.Model):
+class Report(models.Model):
+    """Reporting an article model"""
+    body = models.TextField()
+    author = models.ForeignKey('authentication.User', on_delete=models.CASCADE)
```
Think you should define as reporter instead of author

AnguleMathias replied:
resolved

AnguleMathias committed:
Feature(Article Reports): Users should be able to report an article

- Feedback implementation

[Feature #16125537]

AnguleMathias committed:
Feature(Article Reports): Users should be able to report an article

- Feedback implementation

[Feature #16125537]

Commit:
#161255376 comment track edit history (#32)

* Feature(track comment history):Comments should be able to track edit history

- added CommentHistory model
- modified comment update to store history

[Feature #161255376]

* Feature(track comment history):Comments should be able to track edit history

    - remove debug statements
    - add docstrings

    [Feature #161255376]

3N61N33R reviewed (approved):
LGTM

AnguleMathias committed:
Feature(Article Reports): Users should be able to report an article

- Feedback implementation render data with article id

[Feature #16125537]

AnguleMathias committed:
Feature(Article Reports): Users should be able to report an article

- Feedback implementation render data with article id

[Feature #16125537]

3N61N33R committed:
#161255369 Add Search Functionality and Custom filtering based on parameters (#28)

* feature(search_filter): search, filter functionality

-add app search and filter functionality

[feature #161255369]

* feature(search_filter): search, filter functionality

-fix list articles bug
-add doctstrings

[feature #161255369]

* feature(filter_search): filter, search functionality

-add docstrings
-remove repeated classes

[feature #161255369]

njeri-ngigi committed:
#161255374 Highlight and Comment on any text in an article (#31)

* Feature(Highlight and comment on text): Add CRUD for highlights and comments on text
- Add CRUD functions for higlighting and commenting on any text of an article
- Add tests for these CRUD functions
[Feature #161255374]

* Feature(Highlight and comment on text): Add CRUD for highlights and comments on text
- Add CRUD functions for higlighting and commenting on any text of an article
- Add tests for these CRUD functions
[Feature #161255374]

* Feature(#161255374 Highlighting and commenting on text): Lint code
- Adjust code to match pep8 linting standards
[Feature #161255374]

njeri-ngigi committed:
#161255370 Users should be able to see the time it takes to read an article (#34)

* Feature(#161255370 Time to read): Calculate time to read
- Add method to calculate time to read an article
- Add test for calculating time to read an article
[Feature #161255370]

* Feature(#161255370 Time to read): Lint code
- Adjust code to match pep8 linting standards
[Feature #161255370]

* Feature(#161255370 Time to read): Lint code
- Adjust code to match pep8 linting standards
[Feature #161255370]

3N61N33R committed:
#161255368 Pagination support for articles (#35)

* feature(pagination): pagination for articles

-enable pagination for articles

[feature #161255368]

* feature(pagination): article pagination functionality

-override pagination response
-fix search and filter functionality
- resolve merge conflicts

[feature #161255368]

* feature(pagination): article pagination functionality

-remove import error

[feature #161255368]

* feature(pagination): article pagination functionality

-fix tests

[feature #161255368]

AnguleMathias committed:
#161255377 Users should be able to like a specific comment (#29)

* Feature(Comment likes):Users should be able to like/dislike comments to articles

- Added comment-likes and comment_dislikes fields to models
- Added comment-likes and comment-dislikes method to comment serializer class
- Automatic updates to  migrations
- Deleted .DS_Store

[Feature #161255377]

* Feature(Comment likes):Users should be able to like/dislike comments to articles

- Added comment-likes and comment_dislikes fields to models tests
- Update comment-likes and comment-dislikes method to comment serializer class
- Updates to migrations

[Feature #161255377]

* #161255376 comment track edit history (#32)

* Feature(track comment history):Comments should be able to track edit history

- added CommentHistory model
- modified comment update to store history

[Feature #161255376]

* Feature(track comment history):Comments should be able to track edit history

    - remove debug statements
    - add docstrings

    [Feature #161255376]

* Feature(Comment likes):Users should be able to like comments to articles

- Feedback implementation

[Feature #161255377]

* #161255369 Add Search Functionality and Custom filtering based on parameters (#28)

* feature(search_filter): search, filter functionality

-add app search and filter functionality

[feature #161255369]

* feature(search_filter): search, filter functionality

-fix list articles bug
-add doctstrings

[feature #161255369]

* feature(filter_search): filter, search functionality

-add docstrings
-remove repeated classes

[feature #161255369]

* #161255374 Highlight and Comment on any text in an article (#31)

* Feature(Highlight and comment on text): Add CRUD for highlights and comments on text
- Add CRUD functions for higlighting and commenting on any text of an article
- Add tests for these CRUD functions
[Feature #161255374]

* Feature(Highlight and comment on text): Add CRUD for highlights and comments on text
- Add CRUD functions for higlighting and commenting on any text of an article
- Add tests for these CRUD functions
[Feature #161255374]

* Feature(#161255374 Highlighting and commenting on text): Lint code
- Adjust code to match pep8 linting standards
[Feature #161255374]

* #161255370 Users should be able to see the time it takes to read an article (#34)

* Feature(#161255370 Time to read): Calculate time to read
- Add method to calculate time to read an article
- Add test for calculating time to read an article
[Feature #161255370]

* Feature(#161255370 Time to read): Lint code
- Adjust code to match pep8 linting standards
[Feature #161255370]

* Feature(#161255370 Time to read): Lint code
- Adjust code to match pep8 linting standards
[Feature #161255370]

* #161255368 Pagination support for articles (#35)

* feature(pagination): pagination for articles

-enable pagination for articles

[feature #161255368]

* feature(pagination): article pagination functionality

-override pagination response
-fix search and filter functionality
- resolve merge conflicts

[feature #161255368]

* feature(pagination): article pagination functionality

-remove import error

[feature #161255368]

* feature(pagination): article pagination functionality

-fix tests

[feature #161255368]

* Feature(Comment likes):Users should be able to like/dislike comments to articles

- Added comment-likes and comment_dislikes fields to models
- Added comment-likes and comment-dislikes method to comment serializer class
- Automatic updates to  migrations
- Deleted .DS_Store

[Feature #161255377]

* Feature(Comment likes):Users should be able to like/dislike comments to articles

- Added comment-likes and comment_dislikes fields to models tests
- Update comment-likes and comment-dislikes method to comment serializer class
- Updates to migrations

[Feature #161255377]

* Feature(Comment likes):Users should be able to like comments to articles

- Feedback implementation

[Feature #161255377]

* Feature(Comment likes):Users should be able to like comments to articles

- Rebase commit

[Feature #161255377]

* Feature(Comment likes):Users should be able to like comments to articles

- Rebase commit

[Feature #161255377]

* Feature(Comment likes):Users should be able to like comments to articles

- Refactor tests

[Feature #161255377]

AnguleMathias committed:
Feature(Article Reports): Users should be able to report an article

- Rebasing

[Feature #16125537]

AnguleMathias committed:
Feature(Article Reports): Users should be able to report an article

- Pull from origin conflict resolution

[Feature #16125537]
